packages/webgpu/cpp/rnwgpu/api/GPUCanvasContext.cpp

```diff
@@ -23,41 +23,101 @@ void GPUCanvasContext::configure(
   if (!conv(surfaceConfiguration.usage, configuration->usage)) {
     throw std::runtime_error("Error with SurfaceConfiguration");
   }
-  surfaceConfiguration.width = _canvas->getWidth();
-  surfaceConfiguration.height = _canvas->getHeight();
-  _instance.Configure(&surfaceConfiguration);
-  _lastConfig = configuration;
-  _width = _canvas->getWidth();
-  _height = _canvas->getHeight();
+  auto width = _canvas->getWidth();
+  auto height = _canvas->getHeight();
+  surfaceConfiguration.width = width;
+  surfaceConfiguration.height = height;
+  _surfaceConfiguration = surfaceConfiguration;
+  if (_instance) {
+    _instance.Configure(&surfaceConfiguration);
+  } else {
+    _offscreenSurface->configure(_surfaceConfiguration);
+    // Add texture to the surface registry, when the native surface is available,
+    // we will copy its content there
+    // This only makes sense if the on screen native surface is not available yet
+    auto &registry = rnwgpu::SurfaceRegistry::getInstance();
+    SurfaceInfo info;
+    info.width = width;
+    info.height = height;
+    info.texture = _offscreenSurface->getCurrentTexture();
+    info.gpu = _gpu->get();
+    info.config = _surfaceConfiguration;
+    registry.addIfEmptySurface(_contextId, info);
+  }
 }
 
 void GPUCanvasContext::unconfigure() {
-  _lastConfig = nullptr;
-  _width = _canvas->getWidth();
-  _height = _canvas->getHeight();
-  _instance.Unconfigure();
+  if (_instance) {
+    _instance.Unconfigure();
```

Comment:
I'm just wondering, what if `_offscreenSurface` was already configured? Should you also unconfigure it here, or is it not necessary?

Reply:
it's not necessary but I will clarify this indeed.